internal/resource/url.go

```diff
@@ -47,70 +51,112 @@ const (
 	oemMountPath  = "/mnt/oem"               // Mountpoint where oem partition is mounted when present.
 )
 
-// FetchConfig fetches a raw config from the provided URL.
-func FetchConfig(l *log.Logger, c *HttpClient, u url.URL) ([]byte, error) {
-	return FetchConfigWithHeader(l, c, u, http.Header{})
+const (
+	CompressionNone = ""
+	CompressionGzip = "gzip"
+)
+
+// Fetcher holds settings for fetching resources from URLs
+type Fetcher struct {
+	Logger *log.Logger
+
+	// Client is the http client that will be used when fetching http(s)
+	// resources. If left nil, one will be created and used, but this means any
+	// timeouts Ignition was configured to used will be ignored.
+	Client *HttpClient
```

Comment:
I'm not sure I follow. Do you mean to make an `HttpFetcher` to use for specifically making HTTP requests? What benefit would that have over just using the generic `Fetcher`? I can't off the top of my head think of any locations in Ignition that explicitly need to force http, beyond just specifying that in the url scheme.